homeassistant/components/tessie/update.py

```diff
@@ -33,6 +37,16 @@ def __init__(
         """Initialize the Update."""
         super().__init__(coordinator, "update")
 
+    @property
+    def supported_features(self) -> UpdateEntityFeature:
+        """Flag supported features."""
+        if self.get("vehicle_state_software_update_status") in (
+            TessieUpdateStatus.AVAILABLE,
+            TessieUpdateStatus.SCHEDULED,
```

Comment:
Does this check mean that we only temporarily support install when there's an update available or an update is scheduled or does it mean that vehicles that support available or scheduled updates always support install?

Reply:
The first one.

You can only tell the vehicle to update if it's available and finished downloading. If the update is already scheduled you can install it now.

Reply:
Why don't we just use this logic to tell if an update is available? Why do we need to limit the install feature?

Reply:
Because the user needs to be told if there is an update avaliable, even if its not ready to install right now. For example, if they park in an underground car park and need to move the vehicle into WiFi range.

Reply:
Why is that important? How does the user know how they should make an update ready to be installed? Isn't it just confusing to sometimes have an update available that can not be installed and sometimes have an update available that can be installed?

Reply:
Are you suggesting the integration should lie to the user, and say that an update isnt avaliable when it is?

I dont think that is confusing, because thats how every operating system update I know about works. You get notified its avaliable, you have to wait for it to download, and then you can install it when suitable. You cant install it if you have no network connectivity or its still being downloaded.

Reply:
Depends how you define available. We can define it as available when it's ready to be installed since this platform supports install.

I don't agree with the description about how OS updates works. Normally you're never shown an update that you can't download or install if the system supports downloading and installing updates.

Reply:
This isn't a perfect example because I forgot to turn my WiFi off before checking for updates, but the whole concept of a pending update that is not yet ready to be installed isn't that foreign.

![Screenshot_20240209-133138.png](https://github.com/home-assistant/core/assets/2647914/531677a0-597a-4edb-baf7-feea89d4b1ab)

Reply:
There are no such instructions here now, right? Nor is there a way to resolve the situation in the user interface for the user. Moving the car isn't something that can be resolved from the user interface.

Is it really that important to show that an update is available for a user such that they would go out and move their car to be able to install it? I'd think it's enough to show the update when it's available for install.

Reply:
Well the Update platform doesnt have a way to communicate this information. As another data point, the car itself, and the Tesla app, will tell the user that an update is avaliable but is waiting for WiFi, so its not like this is secret knowledge that the update has to download before it can be installed, and will only download over WiFi.

As a fictitious but plausable example, as user may use an automation to turn on a wireless hotspot when an update is avaliable, and then disable it again once its ready to install/installed.

Reply:
I don't think this answers my question.

Reply:
So before this PR was merged, was it completely pointless in your viewpoint because it wasn't actionable from the UI?

Are you saying every update platform without the availability to install updates is bad?

I'm frustrated because this implementation is working perfectly. I'd probably prefer to rip out the install functionally (this PR) rather than lie about an update not being available.

Reply:
No, I'm saying the current logic seems confusing for the user, as I've already written above.

Reply:
I think the opposite, If the Tessie app is showing the user an update is available, but the Tessie integration isn't, the user will be confused and think the integration is broken and raise an issue.

I will update the integration documentation to make it clear that you can only install updates once they have downloaded.